### DISCUSSION TITLE
brew install xyz throws 401 unauthorised

### DISCUSSION THREAD

**Discussion by Pradumn99** [Homebrew/discussions#3150] [Everyday usage]
Hi,

I am facing a issue that, **while doing brew install xyz it throws 401 unauthorised**. Actually b**rew tries to download xyz which is hosted in Artifactory and is password protected**.

Is there a way t**o pass username and password while doing brew install xyz**.

**Comment by SMillerDev** [comment]
It sounds like you're using a custom bottle domain, maybe one that's issued by a company. Homebrew stopped using artifactory last year so I can't really help you with that.

**Comment by carlocab** [comment]
What is the output of `brew config` and `brew doctor`?

**Reply by Pradumn99** [reply]
**brew config** gives this :

HOMEBREW_VERSION: 3.4.5-16-gf6ab300
ORIGIN: https://github.com/Homebrew/brew
HEAD: f6ab300fc169777705f025c67c4b7c35605f4a66
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a3a7804c5beacc4e44f9dfdd409edbebdfa86514
Core tap last commit: 20 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.35.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6.2-x86_64
CLT: 12.5.1.0.1.1623191612
Xcode: N/A
